packages/generator-particle-base/src/generators/app/generatePromptOptions.ts

```diff
@@ -58,129 +48,88 @@ export const configurationPrompt = [
         return 'Please enter a two word project name with no spaces'
       }
       return true
-    },
+    }
   },
   {
     type: 'input',
     message:
-      'choose a component library name using kebab case. example "project-default"',
-    name: 'componentLibraryName',
-    default: 'project-default',
+      'Where does your design theme exist relative to the root of the project',
+    default: (answers: DesignTheme) =>
+      `./source/design/${answers.themeName}/`,
+    name: 'themePath'
+  }
+]
+
+const compileThemes = async (prev: DesignTheme[]) => {
+  return [...prev, await inquirer.prompt(designThemePrompt())]
+}
+
+const rerun = [{
+  type: 'confirm',
+  name: 'generateTheme',
+  message: 'Do you want to add another theme?',
+  default: true
+}]
+
+export const generatorLoop = async() => {
+  let loop = true;
+  let themesArray:DesignTheme[] = []
+  do {
+      themesArray = await compileThemes(themesArray)
+      loop = await inquirer.prompt(rerun).then(answers => answers.generateTheme)
+    } while (loop)
+
+  return themesArray;
+}
+
+export const propOptions = [
+  {
+    type: 'input',
+    message: 'Choose a abbreviation for your/client\'s name. (min 3 chars)',
+    name: 'clientAbbreviation',
     validate: (name: string) => {
-      if (!name || name.length < 4) {
-        return 'Please enter a library name of more than 4 characters length'
+      if (!name || name.length < 3) {
+        return 'Please enter a project name of more than 4 characters length.'
+      }
+      if (name.indexOf(' ') > 0) {
+        return 'Please enter a two word project name with no spaces.'
```

Comment:
We're repeating this line below and the logic does not match the returned message. It should be `name.length < 4`. Can we create a logic block that reuses this.

example return { isValid: boolean, message: string}
```
const validateNameLength = (name:string, minLength: number) => {
 if (name && name.length >= minLength){
  return { isValid: true }
}
return { isValid: false, message:  `Please enter a project name of more than ${minLength} characters length.`}
}
```

Reply:
See 105 for the repeat logic.

Reply:
We may just be able to reuse the entire validation function. Up to you. Just want to clean up duplicates.